ldap/servers/slapd/log.c

```diff
@@ -156,6 +156,22 @@ get_syslog_loglevel(int loglevel)
     return default_level;
 }
 
+static void
+compress_log_file(char *filename)
+{
+    /* Compress a rotated log */
+    char cmd[BUFSIZ] = {0};
+
+    PR_snprintf(cmd, sizeof(cmd), "/usr/bin/gzip %s", filename);
+    if (PR_SUCCESS != PR_Access(filename, PR_ACCESS_EXISTS)) {
+        PR_snprintf(cmd, sizeof(cmd), "Can not compress log file because it does not exist (%s)\n", filename);
+        log__error_emergency(cmd, 0, 0);
+        return;
+    }
+    /* gzip it */
+    system(cmd);
```

Comment:
I am a bit wary about using system with ns-slapd. I wonder if we should not use zlib instead ?

Reply:
zlib!!  Didn't know about zlib.  :-)  I also did not want to use system(), but I didn't know there was an alternative.  So I will definitely look into this.  This will also remove the selinux issue I ran into.